### PR TITLE
enh/create set

### PR DESCRIPTION
**Pull request by BardiyaAk**
There was a bug in concatenating the sets generated (we were only keeping the last set). The section was in the loop, brought it out to fix the issue.

Refactored the set generation code slightly to make it more granular and more testable. Added some tests to check for bugs and issues.

### CODE REVIEW

File: setlexsem/generate/generate_sets.py [hunk 1]
```diff
@@ -44,28 +45,69 @@ def get_parser():
...
+
+
 def make_sets_from_sampler(
     sample_set: Sampler,
     num_runs: int,
 ) -> List[Dict[str, Any]]:
     """Generate random sets from the sampler"""
 
     # initlize the dataset
+    empty_sampler_counter = 0
```
Comment (ndronen) [inline setlexsem/generate/generate_sets.py]: Probably `empty_sample_count`?

File: setlexsem/generate/generate_sets.py [hunk 2]
```diff
@@ -44,28 +45,69 @@ def get_parser():
...
+            {
+                "experiment_run": i,
+                "A": A,
+                "B": B,
+            }
+        )
+
+    if empty_sampler_counter >= 10:
+        logger.warning(
+            f"Sampler is empty {empty_sampler_counter} times out of {num_runs}"
```
Comment (ndronen) [inline setlexsem/generate/generate_sets.py]: `f"Sampled at least one empty set {empty_sample_count}  out of {num_runs} samples."

File: setlexsem/generate/generate_sets.py [hunk 3]
```diff
@@ -44,28 +45,69 @@ def get_parser():
...
+        set_a = astype_set(raw_set_a)
+        set_b = astype_set(raw_set_b)
+        return set_a, set_b
+    except (ValueError, SyntaxError) as e:
+        logger.error(f"Failed to parse sets: {e}")
+        raise
+
+
+def generate_set_pair(sampler: Union[Iterable, callable]) -> Tuple[set, set]:
+    """Generate a pair of sets from either an iterable or callable sampler"""
```
Comment (ndronen) [inline setlexsem/generate/generate_sets.py]: Can you add more to this doc string that explains, to the user, the motivation for this function?

**Review by ndronen — COMMENTED** [review]
Some nits and a request for a longer doc string.

**Comment by BardiyaAk** [comment]
Addressed comments

**Review by ndronen — APPROVED** [review]
LGTM

**Review by phenomenal-manish — APPROVED** [review]
LGTM!

**Pull request merged**